### DISCUSSION TITLE
Automation api with (un)packaged winui3 app not working

### DISCUSSION THREAD

**Discussion by Gitii** [microsoft/WindowsAppSDK#2745] [Q&A] [answered]
Hi, 

I try to add end to end tests to my project.
It's a winui3 app that is packaged by default.
When using `inspect`, the element `DesktopChildSiteBridge` has no children and thus I cannot use the automation api to write tests using appium.
I also tried it with an unpackaged version of my app.
This time, whenever `inspect` is running my app freezes after startup and then crashes.
Strangly debugging in Visual Studio "prevents" a crash but  `DesktopChildSiteBridge` is still empty.

So does winui3 / windowsappsdk 1.X.X support the automation api?
Or have I done anything wrong?

Best regards,

Gitii

**Comment by castorix** [comment]
It works for me with random controls :
(Windows 10 21H1 (french), Windows App SDK 1.1.0, Unpackaged, C#, test with Inspect x64)

![image](https://user-images.githubusercontent.com/22345506/179405518-eff9ceb2-4903-4fb8-acbe-a82296f519b6.png)

**Reply by Gitii** [reply]
Thanks for your test run, @castorix !
I already assumed that it might be something in my app that breaks the automation api.
Thing is that I have no idea why 🤷‍♂️
Does it work for you when it's packaged?

**Reply by castorix** [reply]
> Does it work for you when it's packaged?

Yes, no difference.

**Reply by Gitii** [reply]
Thanks, I was able to solve the issue:
The main thread wasn't a STA thread.
See https://github.com/sharpninja/WindowsAppSdkHost/issues/3 for details

**Comment by Gitii** [comment, marked as answer]
Solution: The main thread wasn't a STA thread.
See https://github.com/sharpninja/WindowsAppSdkHost/issues/3 for details
